File: apps/web/types/document.ts
export type DocumentStatus =
  | "uploading"
  | "processing"
  | "parsing"
  | "chunking"
  | "embedding"
  | "indexing"
  | "completed"
  | "failed";

export interface Document {
  id: string;
  filename: string;
  document_type: "pdf" | "docx" | "txt";
  content_type: string;
  size_bytes: number;
  status: DocumentStatus;
  error_message: string | null;
  page_count: number | null;
  owner_id: string;
  created_at: string;
  updated_at: string;
}

export interface DocumentListResponse {
  documents: Document[];
}

export const TERMINAL_STATUSES: DocumentStatus[] = ["completed", "failed"];

File: apps/api/app/schemas/document.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: uuid.UUID
    filename: str
    document_type: str
    content_type: str
    size_bytes: int
    status: str
    error_message: str | None
    page_count: int | None
    owner_id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DocumentListResponse(BaseModel):
    documents: list[DocumentResponse]
